File: examples/cn_example.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
中国空气质量指数（AQI）计算示例

运行: 在项目根目录执行 `uv run python examples/cn_example.py`
"""

from aqi_hub.aqi_cn.aqi import (
    AQI,
    cal_aqi_cn,
    cal_iaqi_cn,
    cal_primary_pollutant,
    get_aqi_level,
    get_aqi_level_color,
)


def main() -> None:
    print("=== 中国 AQI 示例 ===\n")

    # 1. AQI 计算
    print("1. AQI 计算")
    aqi, iaqi = cal_aqi_cn(
        pm25=45, pm10=80, so2=35, no2=85, co=3, o3=140, data_type="hourly"
    )
    print("  小时值: AQI =", aqi, ", IAQI =", iaqi)

    aqi2, iaqi2 = cal_aqi_cn(
        pm25=120, pm10=180, so2=65, no2=150, co=8, o3=200, data_type="daily"
    )
    print("  日均值: AQI =", aqi2, ", IAQI =", iaqi2)

    # 2. 单项 IAQI
    print("\n2. 单项 IAQI")
    pm25_iaqi = cal_iaqi_cn("PM25_24H", 120)
    pm10_iaqi = cal_iaqi_cn("PM10_24H", 180)
    print("  PM25_24H 120 μg/m³ → IAQI =", pm25_iaqi)
    print("  PM10_24H 180 μg/m³ → IAQI =", pm10_iaqi)

    # 3. AQI 等级
    print("\n3. AQI 等级 (1–6)")
    level = get_aqi_level(120)
    print("  AQI 120 → 等级", level)

    # 4. 等级颜色
    print("\n4. 等级颜色")
    color = get_aqi_level_color(1, "RGB")
    print("  等级 1 RGB:", color)
    color = get_aqi_level_color(3, "RGB_HEX")
    print("  等级 3 RGB_HEX:", color)
    color = get_aqi_level_color(4, "CMYK_HEX")
    print("  等级 4 CMYK_HEX:", color)

    # 5. 首要污染物
    print("\n5. 首要污染物")
    primary = cal_primary_pollutant(iaqi2)
    print("  IAQI 最大时首要污染物:", primary)

    # 6. AQI 类（一站式）
    print("\n6. AQI 类")
    aqi_obj = AQI(
        pm25=120,
        pm10=180,
        so2=65,
        no2=150,
        co=1.0,
        o3=200,
        data_type="daily",
    )
    print("  AQI:", aqi_obj.AQI)
    print("  IAQI:", aqi_obj.IAQI)
    print("  首要污染物:", aqi_obj.primary_pollutant)
    print("  AQI 等级:", aqi_obj.aqi_level)
    print("  等级颜色 RGB:", aqi_obj.aqi_color_rgb)
    print("  等级颜色 RGB_HEX:", aqi_obj.aqi_color_rgb_hex)

    print("\n=== 示例结束 ===")


if __name__ == "__main__":
    main()

File: examples/cn_example.rs
//! 中国空气质量指数（AQI）计算示例
//!
//! 运行: `cargo run --example cn_example`（在项目根目录）

use aqi_hub::{
    cal_aqi_cn, cal_iaqi_cn, cal_primary_pollutant_cn, cn_item_from_str, get_aqi_level_cn,
    get_aqi_level_color_cn, DataType,
};

fn main() {
    println!("=== 中国 AQI 示例 ===\n");

    // 1. AQI 计算
    println!("1. AQI 计算");
    let (aqi, iaqi) = cal_aqi_cn(
        Some(45.0),
        Some(80.0),
        Some(35.0),
        Some(85.0),
        Some(3.0),
        Some(140.0),
        DataType::Hourly,
    );
    println!(
        "  小时值: AQI = {:?}, IAQI = pm25:{:?} pm10:{:?} ...",
        aqi, iaqi.pm25, iaqi.pm10
    );

    let (aqi2, iaqi2) = cal_aqi_cn(
        Some(120.0),
        Some(180.0),
        Some(65.0),
        Some(150.0),
        Some(8.0),
        Some(200.0),
        DataType::Daily,
    );
    println!(
        "  日均值: AQI = {:?}, IAQI = pm25:{:?} pm10:{:?}",
        aqi2, iaqi2.pm25, iaqi2.pm10
    );

    // 2. 单项 IAQI
    println!("\n2. 单项 IAQI");
    let pm25_iaqi = cal_iaqi_cn(cn_item_from_str("PM25_24H").unwrap(), 120.0);
    let pm10_iaqi = cal_iaqi_cn(cn_item_from_str("PM10_24H").unwrap(), 180.0);
    println!("  PM25_24H 120 μg/m³ → IAQI = {:?}", pm25_iaqi);
    println!("  PM10_24H 180 μg/m³ → IAQI = {:?}", pm10_iaqi);

    // 3. AQI 等级
    println!("\n3. AQI 等级 (1–6)");
    let level = get_aqi_level_cn(120);
    println!("  AQI 120 → 等级 {:?}", level);

    // 4. 等级颜色
    println!("\n4. 等级颜色");
    if let Some(c) = get_aqi_level_color_cn(1, "rgb") {
        println!("  等级 1 RGB: {:?}, RGB_HEX: {}", c.rgb, c.rgb_hex);
    }
    if let Some(c) = get_aqi_level_color_cn(4, "rgb") {
        println!("  等级 4 RGB: {:?}, RGB_HEX: {}", c.rgb, c.rgb_hex);
    }

    // 5. 首要污染物
    println!("\n5. 首要污染物");
    let primary = cal_primary_pollutant_cn(&iaqi2);
    println!("  IAQI 最大时首要污染物: {:?}", primary);

    println!("\n=== 示例结束 ===");
}
